File: java/chaincue-real-estate/src/test/java/com/chaincuerealestate/chaincuerealestate/routes/home/HomePageTest.java
package com.chaincuerealestate.chaincuerealestate.routes.home;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.http.ResponseEntity;
import org.springframework.test.context.ActiveProfiles;

import static org.junit.jupiter.api.Assertions.*;

@SpringBootTest
@ActiveProfiles("test")
class HomePageTest {
    @Autowired
    HomePage homePage;

    @BeforeEach
    void setUp() {
    }

    @Test
    void homePage() {
        var dtoResponseEntity = homePage.homePage("token").getBody();

        assertNotNull(dtoResponseEntity);
        assertTrue(dtoResponseEntity.countries().length > 0);
        assertTrue(dtoResponseEntity.recentlyAddedHouses().length > 0);
    }
}

File: kotlin/chaincue-real-estate-reactive/src/test/kotlin/com/chaincuerealestate/chaincuerealestatereactive/routes/home/HomePageTest.kt
package com.chaincuerealestate.chaincuerealestatereactive.routes.home

import kotlinx.coroutines.runBlocking
import org.junit.jupiter.api.RepeatedTest
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.test.context.ActiveProfiles

@SpringBootTest
@ActiveProfiles("test")
class HomePageTest {

    @Autowired
    private lateinit var homePage: HomePage

    @RepeatedTest(100)
    fun `should return home page DTO`()  {
        runBlocking {
            homePage.homePage()
        }
    }
}
